compiler-core/src/language_server/code_action.rs

```diff
@@ -909,3 +909,449 @@ pub fn code_action_add_missing_patterns(
+    line_numbers: LineNumbers,
+    qualified_constructor: Option<QualifiedConstructor<'a>>,
+    edits: Vec<TextEdit>,
+    should_stop: bool,
+    second_pass: bool,
+}
+
+struct QualifiedConstructor<'a> {
+    hover_location: SrcSpan,
+    import: &'a ast::Import<EcoString>,
+    module_aliased: bool,
+    used_name: EcoString,
+    constructor: EcoString,
+    is_type: bool,
+}
+
+impl<'a> QualifiedToUnqualifiedImport<'a> {
+    pub fn new(module: &'a Module, params: &'a CodeActionParams) -> Self {
+        let line_numbers = LineNumbers::new(&module.code);
+        Self {
+            module,
+            params,
+            line_numbers,
+            edits: vec![],
+            qualified_constructor: None,
+            should_stop: false,
+            second_pass: false,
+        }
+    }
+
+    pub fn code_actions(mut self) -> Vec<CodeAction> {
+        self.visit_typed_module(&self.module.ast);
+        self.second_pass = true;
+        self.visit_typed_module(&self.module.ast);
+        if self.edits.is_empty() {
+            return vec![];
+        }
+        let mut action = Vec::with_capacity(1);
+        CodeActionBuilder::new("Convert to unqualified import")
+            .kind(CodeActionKind::REFACTOR)
+            .changes(self.params.text_document.uri.clone(), self.edits)
+            .preferred(false)
+            .push_to(&mut action);
+        action
+    }
+
+    fn get_module_import(&mut self, module_name: &EcoString) -> Option<&'a ast::Import<EcoString>> {
+        self.module
+            .ast
+            .definitions
+            .iter()
+            .find_map(|def| match def {
+                ast::Definition::Import(import) if import.module == *module_name => Some(import),
+                _ => None,
+            })
+    }
+
+    fn find_in_code(
+        &self,
+        start: usize,
+        end: usize,
+        predicate: impl Fn(&str) -> Option<usize>,
+    ) -> Option<usize> {
+        self.module
+            .code
+            .get(start..end)
+            .and_then(predicate)
+            .map(|pos| start + pos)
+    }
+
+    fn find_trailing_comma_before_closing_brace(
+        &self,
+        import_start: usize,
+        import_end: usize,
+    ) -> Option<usize> {
+        let import_code = self.module.code.get(import_start..import_end)?;
+        let closing_brace_pos = import_code.rfind('}')?;
+
+        let bytes = import_code.as_bytes();
+        let mut pos = closing_brace_pos;
+        while pos > 0 {
+            pos -= 1;
+            let c = (*bytes.get(pos)?) as char;
+            if c.is_whitespace() {
+                continue;
+            }
+            if c == ',' {
+                return Some(import_start + pos);
+            }
+            break;
+        }
+        None
+    }
+
+    fn edit_import(&mut self, name: String) {
+        let qualified_constructor = self
+            .qualified_constructor
+            .as_ref()
+            .expect("import should be set");
```

Comment:
Is this expectation correct? It says import but the property is called constructor.